src/java/org/apache/cassandra/schema/SchemaConstants.java

```diff
@@ -77,14 +78,32 @@ public final class SchemaConstants
      */
     public static final int NAME_LENGTH = 48;
 
+    /**
+     * Longest acceptable file name. Longer names lead to too long file name error.
+     */
+    public static final int FILENAME_LENGTH = 255;
+
+    /**
+     * Longest acceptable table name, so it can be used in a directory
+     * name constructed with a suffix of a table id and a separator.
+     */
+    public static final int TABLE_NAME_LENGTH = FILENAME_LENGTH - 32 - 1;
+
     // 59adb24e-f3cd-3e02-97f0-5b395827453f
     public static final UUID emptyVersion;
 
     public static final List<String> LEGACY_AUTH_TABLES = Arrays.asList("credentials", "users", "permissions");
 
-    public static boolean isValidName(String name)
+    /**
+     * Validates that a name contains only alphanummeric characters or underscore,
+     * so it can be used in file or directory names.
+     *
+     * @param name the name to check
+     * @return whether the name contains only valid characters
```

Comment:
note: we also check that it is not empty

Reply:
The comment at the beginning says:
>a name contains only alphanumeric characters or underscore

Isn't it enough? Or do you want to have it explicit that the name cannot be empty?